daemon/src/lib.rs

```diff
@@ -50,12 +51,18 @@ const HEARTBEAT_INTERVAL: std::time::Duration = Duration::from_secs(5);
 
 pub const N_PAYOUTS: usize = 200;
 
+/// Struct controlling the lifetime of the async tasks,
+/// such as running actors and periodic notifications.
+/// If it gets dropped, all tasks are cancelled.
+pub struct Tasks(Vec<RemoteHandle<()>>);
+
 pub struct MakerActorSystem<O, M, T, W> {
     pub cfd_actor_addr: Address<maker_cfd::Actor<O, M, T, W>>,
     pub cfd_feed_receiver: watch::Receiver<Vec<Cfd>>,
     pub order_feed_receiver: watch::Receiver<Option<Order>>,
     pub update_cfd_feed_receiver: watch::Receiver<UpdateCfdProposals>,
     pub inc_conn_addr: Address<T>,
+    pub tasks_handle: Tasks,
```

Comment:
Just `tasks` perhaps?

Reply:
Plus make it private? Should not be modified I guess.

Reply:
renamed. 

 tried to make it private in the very beginning, but it was getting dropped if I didn't destructure it properly and keep it in scope.

An alternative would be manage the whole actor system sin Rocket, but we're not doing right now (should we?). Then it could be private, as it should be!

That's the reason I made the newtype, which at least makes the contents of `Tasks` private, so you can't add/remove/access the internals.

Reply:
> tried to make it private in the very beginning, but it was getting dropped if I didn't destructure it properly and keep it in scope.

Ah fair enough.

> An alternative would be manage the whole actor system sin Rocket, but we're not doing right now (should we?). Then it could be private, as it should be!

Yeah that might be worth it at some point. Could even consider making all fields private and only expose what we need.

Reply:
I'll put it on my list to deal with at some point.